sycl/include/sycl/ext/oneapi/work_group_static.hpp

```diff
@@ -0,0 +1,61 @@
+//==----- work_group_static.hpp --- SYCL group local memory extension -----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include <sycl/detail/defines_elementary.hpp> // for __SYCL_ALWAYS_INLINE
+#include <sycl/exception.hpp>                 // for exception
+
+#include <type_traits> // for enable_if_t
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi {
+namespace experimental {
+
+#ifdef __SYCL_DEVICE_ONLY__
+#define __SYCL_WG_SCOPE [[__sycl_detail__::wg_scope]]
+#else
+#define __SYCL_WG_SCOPE
+#endif
+
+/// @brief Allocate data in device local memory.
+/// Any work_group_static object will be place in device local memory and hold
+/// an object of type T. work_group_static object are implicitly treated as
+/// static.
+/// @tparam T must be a trivially constructible and destructible type
+template <typename T> class __SYCL_WG_SCOPE work_group_static final {
```

Comment:
FYI @gmlueck @Pennycook, I made the class `final` as I don't think it is a good thing to allow users to derive from it given the implicit behaviour the class has.